homeassistant/components/aurora_abb_powerone/config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Config flow for Aurora ABB PowerOne integration."""
+import logging
+
+from aurorapy.client import AuroraError, AuroraSerialClient
+import serial.tools.list_ports
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_ADDRESS, CONF_PORT
+
+# pylint doesn't think DOMAIN is used.  But it is used!
+# Might be this bug? https://github.com/PyCQA/pylint/issues/3445
+# pylint: disable=unused-import
+from .const import (
+    ATTR_FIRMWARE,
+    ATTR_MODEL,
+    ATTR_SERIAL_NUMBER,
+    DEFAULT_ADDRESS,
+    DEFAULT_INTEGRATION_TITLE,
+    DOMAIN,
+    MAX_ADDRESS,
+    MIN_ADDRESS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_and_connect(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    comport = data[CONF_PORT]
+    address = data[CONF_ADDRESS]
+    _LOGGER.debug("Intitialising com port=%s", comport)
+    ret = {}
+    ret["title"] = DEFAULT_INTEGRATION_TITLE
+    try:
+        client = AuroraSerialClient(address, comport, parity="N", timeout=1)
+        client.connect()
+        ret[ATTR_SERIAL_NUMBER] = client.serial_number()
+        ver = client.version()
+        partnum = client.pn()
```

Comment:
Do these generate I/O?

Reply:
[Yes](https://code.energievalsabbia.it/ccatterina/aurorapy/blob/master/aurorapy/client.py#L241), sends and receives data via the serial port.

Reply:
@bdraco Is there a restriction on doing IO here?

If so, what is the preferred way of doing IO during a config flow?

Reply:
Wrap it in `await hass.async_add_executor_job()`

Reply:
Done